akka-stream/src/main/scala/akka/stream/scaladsl/Tcp.scala

```diff
@@ -117,7 +119,7 @@ final class Tcp(system: ExtendedActorSystem) extends akka.actor.Extension {
       interface: String,
       port: Int,
       backlog: Int = 100,
-      options: immutable.Traversable[SocketOption] = Nil,
+      options: immutable.Iterable[SocketOption] = Nil,
```

Comment:
API changes like this makes me super scared

Reply:
Yeah, I realise maybe we can't do this. 

Iterable extends Traversable, but the other way around, it could be possible (but perhaps not likely) that a user had some special Traversable impl that wasn't Iterable.

In general they'd likely didn't specify anything or put a standard collection in here and then I think it'd be safe.

Reply:
The reason for the change was that `Traversable` is deprecated in 2.13

Reply:
it might be source compatible but not binary compatible

Reply:
There is a slightly similar situation with `IterableOnce`, which did not exist at all in 2.12 and replaces `TraversableOnce` in 2.13.

That one is worked around without binary incompatibility by defining a `type IterableOnce[+X] c.TraversableOnce[X]` in `akka.util.ccompat` for versions lower than 2.13 and using that type there, while not defining one in ccompat and using the 'real' `IterableOnce` on 2.13.

I didn't do the same for `immutable.Iterable` because unlike `IterableOnce`, `immutable.Iterable` does exist on 2.12 so that could get pretty confusing... in theory we could introduce a third name that means `Traversable` on 2.12 and `Iterable` on 2.13, but that still seems weird, might be better to just `@silent` this API?

Reply:
Yeah, I'll revert and try to silence instead.